memoryexpt2/experiment.py

```diff
@@ -159,7 +167,14 @@ def bonus(self, participant):
         """Give the participant a bonus for waiting."""
 
         DOLLARS_PER_HOUR = 5.0
-        t = participant.end_time - participant.creation_time
+        DATETIME_FORMAT = "%Y-%m-%d %H:%M:%S"
+        try:
+            end_waiting_room = datetime.datetime.strptime(participant.property1,
+                                                          DATETIME_FORMAT)
+        except ValueError:
+            # just in case something went wrong saving wait room end time
```

Comment:
This seems like a reasonable fallback. Another option might be the median of all Participants who *do* have end times in `property1`, since they should all be leaving the waiting room at the same time(?) Unsure about that.

Reply:
I guess then you'd need a two-tiered fallback, in case none of the P's have recorded times, so maybe that's excessively complicated.

Reply:
Yeah, not sure if it's worth the effort to have more fallback code. I did the restructure and fixed a bug.